vignettes/ConvertingPharmacoSetToGDR.Rmd

```diff
@@ -0,0 +1,107 @@
+With this functionality, users familiar with the gDR suite of packages and methods can utilize the publically available, curated datasets from the PharmacoGx database. 
+The main step in this process is to extract the drug dose-response data from the PharmacoSets and transform them into a `data.frame` that can be used as input for the `gDRcore::runDrugResponseProcessingPipeline`.
+
+
+# Loading a PharmacoSet (PSet)
+Whereas a user might already have a pharmacoset loaded in their R session, if they wish to obtain a different pharmacoset or use the same script in the future we provide a helper function to do so. 
+It helps to have a user directory in which to store all pharmacosets, and by passing this directory into the function as a parameter, the function will also check to see if the PSet exists in the user-defined directory.
+This is to ensure that the PSet is not being re-downloaded if it already has.  
+```{r eval = FALSE}
+pset <- getPSet("Tavor_2020")
+pset
+```
+
+```{r, include = FALSE}
+read_mocked_PSets <- function(canonical = FALSE) {
+  readRDS(
+    system.file("extdata", "data_for_unittests", "PSets.rds", package = "gDRimport")
+  )
+}
+pset <- testthat:::with_mock(
+    `PharmacoGx::availablePSets` = read_mocked_PSets,
+    suppressMessages(getPSet(
+      "Tavor_2020", psetDir = system.file("extdata/pset", package = "gDRimport")
+    ))
+  )
+pset
+```
+
+# Converting PharmacoSet to Dataframe for gDR pipeline
+`PharamcoSets` hold data pertaining to the cell lines (@sample slot), drugs (@treatment slot), and dose response experiments (@treatmentResponse slot). 
+The dose response data is stored in a `treatmentResponseExperiment` object and the function `gDRimport::convert_pset_to_df` extracts this information to 
+build a `dataFrame` that can be used as input to the gDR pipeline. 
+
+```{r}
+# Store treatment response data in df_ 
+df_ <- convert_pset_to_df(pharmacoset = pset)
+str(df_)
+```
+
+# Subsetting to extract relevant information
+Most canonical `PharmacoSets` have data pertaining to many cell lines and their response to many drugs (drug-combination data is available in some but its conversion to gDR is not currently supported).
+As such, in the interest of time and resources, it may be useful to subset the data before providing it as input for the gDR pipeline. 
+
+```{r}
+#Convert df to datatable for subsetting ease
+dt <- data.table::as.data.table(df_)
+
+
+# example subset using only 1 cellLine
+x <- dt[Clid==dt$Clid[1]]
+x
+```
+
+# Running drug response pipeline with data
+The subsetted data can now be used as input for the `gDRcore::runDrugResponseProcessingPipeline()`. The output of this function is a `MultiAssayExperiment` object which can be accessed with `gDRutils::convert_se_assay_to_dt()`
+
+```{r eval = FALSE}
+# RUN DRUG RESPONSE PROCESSING PIPELINE
+se <- gDRcore::runDrugResponseProcessingPipeline(as.data.frame(x))
```

Comment:
Why using `as.data.frame`? In function `gDRcore::runDrugResponseProcessingPipeline` param `x` should be one of `c("data.table", "MultiAssayExperiment")` (and in this example `x` is `data.table`)

Reply:
This is leftover from the previous code, thanks.